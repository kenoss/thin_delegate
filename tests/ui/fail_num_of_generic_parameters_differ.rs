#[thin_delegate::register]
pub trait Hello<T> {
    fn hello(&self) -> String;
}

impl<T> Hello<T> for String {
    fn hello(&self) -> String {
        format!("hello, {self}")
    }
}

impl<T> Hello<T> for char {
    fn hello(&self) -> String {
        format!("hello, {self}")
    }
}

#[thin_delegate::register]
enum Hoge {
    A(String),
    B(char),
}

#[thin_delegate::derive_delegate]
impl Hello<u8, u16> for Hoge {}

fn main() {}
