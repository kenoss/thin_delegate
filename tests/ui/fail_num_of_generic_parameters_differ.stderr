error: number of generic parameters must coinside:
           in definition of trait         = < T >
           in definition of derive target = < u8, u16 >
  --> tests/ui/fail_num_of_generic_parameters_differ.rs:25:11
   |
25 | impl Hello<u8, u16> for Hoge {}
   |           ^^^^^^^^^
